### PR TITLE
[3] Add RulesToAdditionalTextConverter

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Part of #4

### CODE REVIEW

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Xml.Serialization;
+
+namespace SonarLint.VisualStudio.Core.CSharpVB
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfiguration.cs:24]: Copied as-is from SLVS

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfigurationSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+namespace SonarLint.OmniSharp.Plugin.Rules.SonarLintXml
+{
+    internal interface ISonarLintConfigurationSerializer
+    {
+        string Serialize(SonarLintConfiguration sonarLintConfiguration);
+    }
+    
+    internal class SonarLintConfigurationSerializer : ISonarLintConfigurationSerializer
+    {
+        public string Serialize(SonarLintConfiguration sonarLintConfiguration)
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfigurationSerializer.cs:36]: Copied from SLVS: https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Core/CSharpVB/Serializer.cs#L55

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/RulesToAdditionalTextConverter.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        internal RulesToAdditionalTextConverter(IRulesToSonarLintConfigurationConverter rulesToSonarLintConfigurationConverter,
+            ISonarLintConfigurationSerializer sonarLintConfigurationSerializer)
+        {
+            this.rulesToSonarLintConfigurationConverter = rulesToSonarLintConfigurationConverter;
+            this.sonarLintConfigurationSerializer = sonarLintConfigurationSerializer;
+        }
+
+        public AdditionalText Convert(IEnumerable<RuleDefinition> rules)
```
Comment (rita-sonarsource) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/RulesToAdditionalTextConverter.cs:54]: The `ICsDiagnosticWorker` will new-up this class (or maybe MEF-import it, we'll see) and call the convert method.
We could then remove all the AdditionalFiles whose path ends with same file name as the returned AdditionalText. This way this is the only place that knows about the `SonarLint.xml` name.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfiguration.cs (hunk 1 above)
Comment (duncanp-sonar) [reply omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/SonarLintXml/SonarLintConfiguration.cs:24]: Worth adding a comment that this file must be kept in step with the format expected by the sonar-dotnet analyzers, and maybe include a link to the corresponding sonar-dotnet class, if there is one.

File: omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/RulesToAdditionalTextConverter.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+namespace SonarLint.OmniSharp.Plugin.Rules
+{
+    internal interface IRulesToAdditionalTextConverter
+    {
+        AdditionalText Convert(IEnumerable<RuleDefinition> rules);
+    }
+
+    internal class RulesToAdditionalTextConverter : IRulesToAdditionalTextConverter
+    {
+        private static readonly string DummySonarLintXmlFilePath = Path.Combine(Path.GetTempPath(), "SonarLint.xml");
```
Comment (duncanp-sonar) [inline omnisharp-dotnet/src/SonarLint.OmniSharp.Plugin/Rules/RulesToAdditionalTextConverter.cs:37]: Worth commenting that this file will never be written to disc so the path is irrelevant.

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**